src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -793,13 +797,69 @@ public void watchFiles(File buildFile, File outputDirectory, File testOutputDire
                     SensitivityWatchEventModifier.HIGH);
             debug("Watching build file directory: " + buildFile.getParentFile().toPath());
 
+            List<File> recompileJavaSources = new ArrayList<File>();
+            List<File> recompileJavaTests = new ArrayList<File>();
+            List<File> deleteJavaSources = new ArrayList<File>();
+            List<File> deleteJavaTests = new ArrayList<File>();
+            long lastJavaSourceChange = System.currentTimeMillis();
+            long lastJavaTestChange = System.currentTimeMillis();
+
             while (true) {
 
                 // stop dev mode if the server has been stopped by another process
                 if (serverThread.getState().equals(Thread.State.TERMINATED) && (this.devStop.get() == false)) {
                     throw new PluginScenarioException("The server has stopped. Exiting dev mode.");
                 }
 
+                // process java source files if no changes detected after the compile wait time
+                boolean processSources = System.currentTimeMillis() > lastJavaSourceChange + compileWaitMillis;
```

Comment:
should this call `this.compileWaitMillis`

Reply:
It doesn't need `this.` since the variable is not ambiguous in this scope.